src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/ComparisonUtils.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    public static class ComparisonUtils
+    {
+        /// <summary>
+        /// Signfies to <see cref="CheckEquality{T}(T, T)"/> that it should use a special <see cref="IEqualityComparer{object}"/> for this property.
+        /// </summary>
+        public class CustomEqualityComparerAttribute : Attribute
+        {
+            /// <param name="type">
+            /// Must be a <see cref="IEqualityComparer{object}"/> with a default constructor.
+            /// </param>
+            public CustomEqualityComparerAttribute(Type type)
+            {
+                if (type == null || !typeof(IEqualityComparer<object>).IsAssignableFrom(type) || type.GetConstructor(new Type[] { }) == null)
+                {
+                    throw new ArgumentException($"{nameof(type)} must be a {nameof(IEqualityComparer<object>)} with a default constructor!");
+                }
+
+                _type = type;
+            }
+
+            private Type _type;
+
+            public IEqualityComparer<object> GetEqualityComparer()
+            {
+                return _type
+                    .GetConstructor(new Type[] { })
+                    .Invoke(new object[] { }) 
+                    as IEqualityComparer<object>;
+            }
+        }
+
+        /// <summary>
+        /// Compares <paramref name="v2"/> and <paramref name="v3"/>.
+        /// </summary>
+        /// <returns>
+        /// If both objects are perfectly equal, returns null.
+        /// If one object is null and the other is not, returns an empty <see cref="IEnumerable{PropertyInfo}"/>.
+        /// Otherwise, returns an <see cref="IEnumerable{PropertyInfo}"/> containing all properties that are not equal.
+        /// </returns>
+        public static IEnumerable<PropertyInfo> CheckEquality<T>(T v2, T v3) where T : class
```

Comment:
Done--split into `Validator`s that exclusively check the properties we care about--which is currently  `Listed`, `PackageContent`, `Id`, `Version`, and `RequireLIcenseAcceptance`. also added a `Validator` that checks if the registration pages exist.